lib/routes/lnurl/lnurl_invoice_delegate.dart

```diff
@@ -72,14 +81,47 @@ Future<LNURLPaymentPageResult?> handlePayRequest(
       ),
     );
   }
-
-  if (pageResult == null) {
+  if (paymentInfo == null) {
     return Future.value();
   }
-  if (pageResult.hasError) {
-    throw pageResult.errorMessage;
-  }
-  return pageResult;
+  // Artificial wait for UX purposes
+  await Future.delayed(const Duration(milliseconds: 800));
+  // Show Processing Payment Dialog
+  // ignore: use_build_context_synchronously
+  return await showDialog(
+    useRootNavigator: false,
+    context: context,
+    barrierDismissible: false,
+    builder: (_) => ProcessingLNURLPaymentDialog(
+      firstPaymentItemKey: firstPaymentItemKey,
+      paymentFunc: () => accountBloc.lnurlPay(
+        amount: paymentInfo!.amount,
+        comment: paymentInfo.comment,
+        reqData: reqData,
+      ),
+    ),
+  ).then((result) {
+    if (result is LnUrlPayResult) {
+      if (result is sdk.LnUrlPayResult_EndpointSuccess) {
+        _log.v("LNURL payment success, action: ${result.data}");
+        return LNURLPaymentPageResult(
+          successAction: result.data,
+        );
+      } else if (result is sdk.LnUrlPayResult_EndpointError) {
+        _log.v("LNURL payment failed: ${result.data.reason}");
+        return LNURLPaymentPageResult(
+          error: result.data.reason,
+        );
+      }
+    } else if (result is FfiException) {
```

Comment:
Perhaps it is better to check for any exception here, not only ffi?

Reply:
Our extractExceptionMessage logic are suited for FfiException's and looking at the previous logic I assumed we returned a generic error for other unknown errors.

Reply:
Yes but the problem it is now depends on ffi here at the high level code which doesn't seem clean.
Maybe it is better to make it generic?
For example we can add optional default error parameter to the  `extractExceptionMessage` in case it fails to find what it is looking for (such as ffiexception).
Something like:
`extractExceptionMessage(Object exception, String defaultErrorMsg)`
Or we can find other pattern as long as we don't handle this detail in high level.

Reply:
Done 90ba55faf48ab9125368595a6f98a50e1452aa26